index.html

```diff
@@ -304,6 +304,10 @@ <h2>
             </li>
           </ul>
         </li>
+        <li>Otherwise, if |manifest|["{{WebAppManifest/start_url}}"] is not
+        present, or not <a>same origin</a> as the {{Document}}'s
+        {{Document/URL}}, consider the document not <a>installable</a>.
```

Comment:
Given that user agents let you install whatever you want (via "add to home screen") - maybe we should say:  
```suggestion
        {{Document/URL}}, a user agent MAY consider the document not <a>installable</a>.
```

Reply:
I like this suggestion. Edge also has the "Install Site as an App" (or similar) menu option for what were "Pinned Sites".

Reply:
Ah, I didn't see this before. It's not so simple that we can simply say "MAY" here; otherwise that would allow user agents to install a document linking a manifest whose `start_url` and `scope` are on a different origin (where previous to this PR, we reset the `start_url` and `scope` to the document URL).

So I think if we change this, we need to have some wording similar to the "non-fatal" case of step 2 above, where we say "the user agent MAY either: 1. set `start_url` to the document URL and `scope` to match, or 2. consider the document not installable."

You're right, Chromium browsers allow this, but I'm sure we don't just let you install a manifest that has its start and scope on another origin. I'll figure out what they do.

Reply:
In Chromium we currently enforce that `start_url` and `scope` are same-origin with the document URL, which the spec requires. If that isn't met, we discard the values.

When this happens, Chromium browsers use the document URL in place of the manifest values when a site has a shortcut created for it. That's behaviour outside the scope of the manifest spec and a browser choice.

Since this PR removes that same-origin requirement, we would remove that enforcement in our parsing when updating to match this.

Reply:
> In Chromium we currently enforce that start_url and scope are same-origin with the document URL, which the spec requires. If that isn't met, we discard the values.

I think treating `start_url` as it if it was not present when it's not same origin or outside the scope makes sense, as does assigning it a sensible default (document URL) when in error. That can then remove the "installable" decision at that point (that shouldn't be part of the processing algorithm, IMO).

Reply:
Right, so that _does_ unfortunately mean that processing the manifest is still a function of the document URL (if we say "discard `start_url` and `scope` if they are not same-origin as document URL and replace them with document URL and the dir of the document URL, respectively"). That's explicitly what I was trying to avoid here.

I think I can live with it if we do that as a separate step outside the "processing the manifest".

That basically means we do "processing the manifest" which is NOT a function of the document URL, and then in a step outside of that, we do the above check if you're trying to install a document.

Marcos, I think you're right, having "installability of a document" doesn't make a lot of sense since user agents can always choose to install any document, and in any case, our current steps to determine installability are MAY MAY MAY. So we can scrap that. But I think we do need to replace this with some other steps for essentially "sanitizing a manifest for installation from a document".

What I want to allow is basically two separate high-level operations:

1. Installing a manifest. (Not associated with a document at all.) That's essentially what's enabled by this PR, by allowing the user agent to process a manifest without a document URL, then install it.
2. Installing a document. Get the associated manifest, if there is one. If there isn't, you can make one up. But this step should ensure that the manifest is same-origin as the document, to prevent sites from being able to install PWAs on another origin.

I propose that 1 is taken care of by just the definition "[install](https://www.w3.org/TR/appmanifest/#dfn-installed)" (what that means is UA-specific). And for 2, we should replace "steps to determine the installability of the document" with the following:

**Steps to install a document**, given a processed manifest _manifest_:
1. Let _manifest_ and _manifest URL_ be the result of [obtaining the manifest](https://www.w3.org/TR/appmanifest/#dfn-obtaining-the-manifest).
2. If obtaining the manifest results in an error, set _manifest_ to a processed manifest, populated from the top-level browsing context Document's metadata in a user-agent-specific way (e.g., setting `manifest["name"]` to the document title).
3. If `manifest["start_url"]` is not present, or not same origin as the Document URL, set `manifest["start_url"]` to the Document URL, and set `manifest["scope"]` to the result of parsing "." using the Document URL as the base URL.
4. [Install](https://www.w3.org/TR/appmanifest/#dfn-installed) the manifest.

This is too much work to do in this PR, so I think I'll just continue propping up the "installability of the document" concept right now, and do the above later after some more discussion.

Reply:
OK, I've updated the PR. I filed a bug #912 about the above issue (wherein Marcos and I agree, we shouldn't have "installability of the document") but I haven't tackled it here.

I've updated the text so the UA can now either consider the document not installable, OR can use the document URL to replace the off-origin start_url.

And I've updated the commit description. Actually, this is no longer a breaking change (!). You can carry on behaving exactly as before, if you like, which is replacing an invalid or off-origin start_url with the document URL, or you can error out (newly allowed behaviour). In one sense, this is now just a (useful) refactor, moving the replace-start-with-document logic up out of the manifest processor and into the "steps to determine installability of the document".

The next step in #912 will be rewriting those steps to be about postprocessing the manifest, not determining installability.

@marcoscaceres how is this?